internal/client/source_test.go

```diff
@@ -183,3 +187,92 @@ func TestGetSourceInfoV0(t *testing.T) {
 	require.NoError(t, err, "DELETE source failed with a client error")
 	require.True(t, resp.Status, "DELETE source failed: %#v", resp)
 }
+
+func UploadUserDefinedSources(t *testing.T, sources []string) {
+	for i := range sources {
+		source := strings.Replace(sources[i], "REPO-PATH", testState.repoDir, 1)
+		resp, err := PostJSONSourceV0(testState.socket, source)
+		require.NoError(t, err, "POST source failed with a client error")
+		require.True(t, resp.Status, "POST source failed: %#v", resp)
+	}
+}
+
+// verify user defined sources are not present
+func VerifyNoUserDefinedSources(t *testing.T, source_names []string) {
+	list, api, err := ListSourcesV0(testState.socket)
+	require.NoError(t, err, "GET source failed with a client error")
+	require.Nil(t, api, "ListSources failed: %#v", api)
+	require.GreaterOrEqual(t, len(list), 1, "Not enough sources returned")
+	for i := range(source_names) {
+		require.NotContains(t, list, source_names[i])
+	}
+}
+
+func TestDeleteUserDefinedSourcesV0(t *testing.T) {
+	source_names := []string{"package-repo-1", "package-repo-2"}
+	sources := []string{`{
+		"name": "package-repo-1",
+		"url": "file://REPO-PATH",
+		"type": "yum-baseurl",
+		"proxy": "https://proxy-url/",
+		"check_ssl": true,
+		"check_gpg": true,
+		"gpgkey_urls": ["https://url/path/to/gpg-key"]
+	}`,
+		`{
+		"name": "package-repo-2",
+		"url": "file://REPO-PATH",
+		"type": "yum-baseurl",
+		"proxy": "https://proxy-url/",
+		"check_ssl": true,
+		"check_gpg": true,
+		"gpgkey_urls": ["https://url/path/to/gpg-key"]
+	}`}
+
+	// verify test starts without user defined sources
+	VerifyNoUserDefinedSources(t, source_names)
+
+	// post user defined sources
+	UploadUserDefinedSources(t, sources)
+	// note: not verifying user defined sources have been pushed b/c correct
+	// operation of PostJSONSourceV0 is validated in the test functions above
+
+	// Remove the test sources
+	for _, n := range(source_names) {
+		resp, err := DeleteSourceV0(testState.socket, n)
+		require.NoError(t, err, "DELETE source failed with a client error")
+		require.True(t, resp.Status, "DELETE source failed: %#v", resp)
+	}
+
+	// verify removed sources are not present after removal
+	VerifyNoUserDefinedSources(t, source_names)
+}
+
+
+func TestDeleteSystemSourcesV0(t *testing.T) {
+	hostDistroName, err := distro.GetHostDistroName()
+	hostArch := common.CurrentArch()
+	require.NoError(t, err)
+
+	// note: this is an integration test and it expects osbuild-composer to be installed
+	// that's why we search for repositories configuration in system wide directories!
+	repoMap, err := rpmmd.LoadRepositories([]string{"/etc/osbuild-composer", "/usr/share/osbuild-composer"}, hostDistroName)
+	require.NoError(t, err, "rpmmd.LoadRepositories: %v", err)
+	require.GreaterOrEqual(t, len(repoMap[hostArch]), 1)
```

Comment:
@bcl after the changes in #404 I'm not sure how these tests are expected to be executed and if repository definitions should be read from the system wide directories. Can you comment on this?

Reply:
After the rebase this now fails b/c I have only a "test-id" repository available and not the fedora and updates repositories, although I am trying to load them.

Reply:
@bcl I am starting to think this better become an integration test. What do you think ?

Reply:
Scratch that. I've managed to rework this test and now it tries to delete the repository which comes configured with the API server.

Reply:
I'm not seeing that commit, it's still trying to delete "test-system-repo" during the integration test.

Reply:
In my previous comment I meant "test-system-repo" (before that I had a different implementation). 

The latest update tries several well known source names and skips them if they aren't available in the initial sources list. That makes the test fault  tolerant and you can execute it both as unit test and as an integration test. In integration test mode I get a failure which doesn't seem related to the current changes.